File: #1 interactive sorting/program.c
#include <stdio.h>
#include <string.h>

// Implicit declaration
int isequal(char e[20], char e1[20], char a[100][20], int n);
int removes(char co[100][20], int n);
void swap(int s, int s1, char a[100][20]);

int main() {

    char a[100][20], b[100][20], sm[100][20];
    char co[200][20];

    int i = 0, j = 0, k = 0;
    
    printf( "!! please press enter key with empty space if given values are enough...");
    printf("\n");
    printf("\n");


    // Get the values until the user stops
    for (; i < 100; i++)
    {
        fgets(a[i], 20, stdin);
        if (a[i][0] == '\n') break;
        a[i][strcspn(a[i], "\n")] = '\0';  // Remove the empty space
    }

    // Making the possible comparisons

    for (; j < i - 1; j++)
    {
        printf("%s ? %s : ", a[j], a[j + 1]);
        char s;
        scanf(" %c", &s);

        if (s == '<')
        {
            strcpy(b[j], a[j + 1]);
            strcpy(sm[j], a[j]);
        } else if (s == '>')
        {
            strcpy(b[j], a[j]);
            strcpy(sm[j], a[j + 1]);
        }
    }

    // Combine both greater and smaller arrays

    for (int m = 0; m < j; m++)
    {
        strcpy(co[k++], b[m]);
    }

    for (int m = 0; m < j; m++)
    {
        strcpy(co[k++], sm[m]);
    }


    int con = removes(co, k);

    // checking the possible comparison in co array if it has already happened or not

    for (int l = 0; l < con - 1; l++)
    {
        if (!isequal(co[l], co[l + 1], a, i))
        {
            printf("%s ? %s : ", co[l], co[l + 1]);
            char s;
            scanf(" %c", &s);
            if (s == '<') {
                swap(l, l + 1, co);
            }
        }
    }

    // Print the final array values

    printf("\n");
    printf("\n Sorting order : ");
    for (int m = 0; m < con; m++)
    {
        printf("%s ", co[m]);
    }

    return 0;
}

int isequal(char e[20], char e1[20], char a[100][20], int n)
{
    for (int j = 0; j < n - 1; j++)
    {
        if ((strcmp(e, a[j]) == 0 && strcmp(e1, a[j + 1]) == 0) ||
                (strcmp(e1, a[j]) == 0 && strcmp(e, a[j + 1]) == 0)) {
            return 1;
        }
    }

    if (strcmp(e, e1) == 0)
    {
        return 1;
    }

    return 0;
}

// Function to remove duplicates from array 'co'

int removes(char co[100][20], int n)
{
    int j, k, l;

    for (j = 0; j < n; j++)
    {
        for (k = j + 1; k < n;)
        {
            if (strcmp(co[j], co[k]) == 0)
            {

                for (l = k; l < n - 1; l++)
                {
                    strcpy(co[l], co[l + 1]);
                }
                
                n--;
            } 
            
            else
            {
                k++;
            }
        }
    }

    return n;  // Return the new size of the array
}

void swap(int s, int s1, char a[100][20])
{
    char temp[20];
    strcpy(temp, a[s]);
    strcpy(a[s], a[s1]);
    strcpy(a[s1], temp);
}

File: #1 interactive sorting/program.java
import java.util.*;

public class program
{
    public static void main(String[] ar) 
    {
        Scanner s=new Scanner(System.in);
        
        System.out.print("!! please press enter key with empty space if given values are enough");
        System.out.println();
        System.out.println();
        
        //define array for getting values
        List <String> a=new ArrayList<>(); 
        
        // define two seperate array for greater and smaller string
        List<String> big=new ArrayList<>();
        List<String> small=new ArrayList<>();
        
        int i=0,l=0,k=0;
        
        //to getting the String arrays until user stops
        
        for(;i<100;i++)
        {
            String a1=s.nextLine();
            a.add(a1);
            if(a1.isEmpty())break;
        }
        
        
        
        for(;k<a.size()-2;k++)
        {
            System.out.print(a.get(k)+" ? "+ a.get(k+1)+": "); //show the possible comparision of two strings 
            char c=s.next().charAt(0);
            if(c=='<')
            {
                small.add(a.get(k)); // add the greater
                
                big.add(a.get(k+1)); // add the smaller
            }
            else if(c=='>')
            {
                small.add(a.get(k+1));
                big.add(a.get(k));
            
            }
        }
        
        //combine both big and small string arrays
        
        List <String> co=new ArrayList<>(big);
        co.addAll(small);
        
        
        
        for(;l<co.size()-1;l++)
        {
            if(!isequal(co.get(l),co.get(l+1),a,co))
               
            {
                System.out.println(co.get(l)+" ? "+co.get(l+1)+": ");
                char c1=s.next().charAt(0);
                if(c1=='<')
                {
                    swap(l,l+1,co); // swap the Strings
                }
            }
        }
        
        //remove the duplicate Strings
        
        Set <String> un=new LinkedHashSet<>(co);
        System.out.println();
        System.out.print("Sorting order : ");
        for(String so:un)
        {
            System.out.print(so+" ");
        }
        
        
    }
    
    static boolean isequal(String e,String e1,List<String> a, List<String> co)
    {
        int i;
        for(i=0;i<a.size()-1;i++)
        {
            if((e.equals(a.get(i)) && e1.equals(a.get(i+1))) || 
                (e1.equals(a.get(i)) && e.equals(a.get(i+1))))
            {
                
                
                return true;
            }
        }
        if(e.equals(e1))
        {
            co.remove(e1);
        return true;
            
        }
        
        return false;
    }
    
   static void swap(int s,int s1,List<String> a)
    {
      String temp=a.get(s);
        a.set(s,a.get(s1));
        a.set(s1,temp);
    }
}